### PR TITLE
Logzio exporter

### PR DESCRIPTION
**Pull request by yyyogev**
**Description:** Added a logz.io traces exporter

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/686

**Testing:** Added test for each of the components in the new exporter

**Documentation:** Added a readme specifying how to use the exporter and its parameters with an example.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@yyyogev please sign the CLA.

**Comment by yyyogev** [comment]
> @yyyogev please sign the CLA.

Will do shortly

File: cmd/otelcontribcol/components.go [hunk 1]
```diff
@@ -15,6 +15,7 @@
 package main
 
 import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/logzioexporter"
```
Comment (rmfitzpatrick) [inline cmd/otelcontribcol/components.go:18]: mind moving this w/ the other exporters (~line 30)?

File: exporter/logzioexporter/config_test.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package logzioexporter
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config/configtest"
+	"path"
+	"testing"
```
Comment (rmfitzpatrick) [inline exporter/logzioexporter/config_test.go:10]: adopting import order conventions would be appreciated:
stdlib
external
collector-contrib local

File: exporter/logzioexporter/testdata/config.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+receivers:
+  examplereceiver:
+
+processors:
+  exampleprocessor:
+
+exporters:
+  logzio:
```
Comment (rmfitzpatrick) [inline exporter/logzioexporter/testdata/config.yaml:8]: I believe the preferred practice is to have multiple component instances, exercising defaults and configuring all possible fields: https://github.com/open-telemetry/opentelemetry-collector/blob/master/exporter/jaegerexporter/testdata/config.yaml#L7

File: exporter/logzioexporter/exporter_test.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+				Name: testService,
+			},
+			Identifier: &commonpb.ProcessIdentifier{
+				HostName: testHost,
+			},
+		},
+		Spans: testSpans,
+	}
+	testTraceExporter(internaldata.OCToTraceData(td), tester, &cfg)
+	//time.Sleep(time.Second * 10)
```
Comment (rmfitzpatrick) [inline exporter/logzioexporter/exporter_test.go:94]: nit: dev remnant

File: exporter/logzioexporter/README.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Logzio Exporter
+
+This exporter supports sending trace data to [Logz.io](https://www.logz.io)
+
+The following configuration options are supported:
+
+* `account_token` (Required): Your logz.io account token for your tracing account.
```
Comment (rmfitzpatrick) [inline exporter/logzioexporter/README.md:7]: If this is required, would it be preferable to check for zero value in factory helper or additional validate() method?

File: exporter/logzioexporter/exporter_test.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+	requests := strings.Split(string(recordedRequests), "\n")
+	var logzioSpan objects.LogzioSpan
+	assert.NoError(tester, json.Unmarshal([]byte(requests[0]), &logzioSpan))
+	assert.Equal(tester, logzioSpan.OperationName, testOperation)
```
Comment (rmfitzpatrick) [inline exporter/logzioexporter/exporter_test.go:98]: nit: second argument for assert helpers should be expected value, followed by actual.

File: exporter/logzioexporter/go.mod [hunk 7]
```diff
@@ -0,0 +1,14 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/exporter/logzioexporter
```
Comment (rmfitzpatrick) [inline exporter/logzioexporter/go.mod:1]: mind committing go.sum as well?

**Comment by rmfitzpatrick** [comment]
@yyyogev it appears the lint and unit test failures are from your addition:
```
#unit:
running go unit test ./... + coverage in /home/circleci/project/exporter/logzioexporter
--- FAIL: TestNullExporterConfig (0.00s)
panic: interface conversion: configmodels.Exporter is nil, not *logzioexporter.Config [recovered]
	panic: interface conversion: configmodels.Exporter is nil, not *logzioexporter.Config

#lint:
exporter.go:6: File is not `goimports`-ed with -local github.com/open-telemetry/opentelemetry-collector-contrib (goimports)
	"fmt"
factory.go:18: File is not `goimports`-ed with -local github.com/open-telemetry/opentelemetry-collector-contrib (goimports)
	"context"
config_test.go:3: File is not `goimports`-ed with -local github.com/open-telemetry/opentelemetry-collector-contrib (goimports)
import (
../../Makefile.Common:110: recipe for target 'lint' failed
make[1]: *** [lint] Error 1
make[1]: Leaving directory '/home/circleci/project/exporter/logzioexporter'
Makefile:91: recipe for target 'for-all-target-./exporter/logzioexporter' failed
make: *** [for-all-target-./exporter/logzioexporter] Error 2
```

**Comment by bogdandrutu** [comment]
Please sign the CLA

**Comment by yyyogev** [comment]
Closing to reopen so the CLA check will run again

**Pull request closed without merging**